### PR TITLE
Refactor CPP Quickstart to use project template project

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

File: docs/tutorial/cpp/src/game_logic_in_cpp.md [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
 
 Insert this code before the `main_window->run()` call:
 
 ```cpp
 {{#include main_game_logic.cpp:game_logic}}
 ```
 
-Notice that we take a weak pointer of our `main_window`. This is very
-important because capturing a copy of the `main_window` itself within the callback handler would result in a circular ownership.
+The code uses a [Weak](https://slint.dev/docs/rust/slint/struct.Weak) pointer of the `main_window`. This is
```
Comment (tronical) [inline docs/tutorial/cpp/src/game_logic_in_cpp.md:43]:
```suggestion
The code uses a [ComponentWeakHandle](https://slint.dev/docs/cpp/api/classslint_1_1ComponentWeakHandle) pointer of the `main_window`. This is
```

File: docs/tutorial/cpp/src/memory_tile.md [hunk 2]
```diff
@@ -1,44 +1,42 @@
...
+following commands:
 
 ```sh
+cd ui
 curl -O https://slint.dev/blog/memory-game-tutorial/icons.zip
 unzip icons.zip
+cd ..
 ```
 
-If you are on Windows, use the following commands:
```
Comment (tronical) [inline docs/tutorial/cpp/src/memory_tile.md:32]: I'm curious, why did you remove the Windows specific instructions?

**Comment by tronical** [comment]
Looks good at first glance. I'll take a closer look tomorrow. The CI is failing because we try to compile the CMakeLists.txt in the source folder to ensure that all the .cpp code used also compiles (we've been bitten by that a few times).

File: docs/tutorial/cpp/src/getting_started.md [hunk 3]
```diff
@@ -1,88 +1,82 @@
...
+Before using the template, you need a C++ compiler that supports C++ 20 and to install [CMake](https://cmake.org/download/) 3.21 or newer.
+
+Clone or download template repository:
+
+```sh
+git clone https://github.com/slint-ui/slint-cpp-template memory
+cd memory
+```
+
+<!-- You will need a development environment that can compile C++20, [CMake 3.21](https://cmake.org/download/),
```
Comment (tronical) [inline docs/tutorial/cpp/src/getting_started.md:19]: Since the old bit is commented out, I guess it would be fine to also remove it entirely :)

File: docs/tutorial/cpp/src/getting_started.md [hunk 4]
```diff
@@ -1,88 +1,82 @@
...
-add_executable(memory_game main.cpp)
-target_link_libraries(memory_game PRIVATE Slint::Slint)
-slint_target_sources(memory_game memory.slint)
-# On Windows, copy the Slint DLL next to the application binary so that it's found.
-if (WIN32)
-    add_custom_command(TARGET memory_game POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_RUNTIME_DLLS:memory_game> $<TARGET_FILE_DIR:memory_game> COMMAND_EXPAND_LISTS)
-endif()
+We're going to use `cmake`'s builtin FetchContent module to fetch the source code of Slint. -->
+
+Change the references to `my_application` in the `CMakeLists.txt` file to `memory`
```
Comment (tronical) [inline docs/tutorial/cpp/src/getting_started.md:27]:
```suggestion
Change all the references to `my_application` in the `CMakeLists.txt` file to `memory`.
```

Comment (tronical) [inline docs/tutorial/cpp/src/getting_started.md:27]: I wonder if the renaming is worth it though - it makes it a little error prone since the instructions just say "change the references".

File: docs/tutorial/cpp/src/game_logic_in_cpp.md (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/game_logic_in_cpp.md:43]: @tronical Sorry, a bit copy/paste happy in a few places, I think 😓

File: docs/tutorial/cpp/src/getting_started.md (hunk 3 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/getting_started.md:19]: @tronical Yeah I kind of left it there in case it turned out later that I needed it after testing, but seemingly not.

File: docs/tutorial/cpp/src/getting_started.md (hunk 4 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/getting_started.md]: @tronical What do you mean? The template project has everything set with different file names, maybe I'm misunderstanding something about C++?

File: docs/tutorial/cpp/src/memory_tile.md (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/memory_tile.md:32]: @tronical I think my reasoning was for a bit of consistency, as we don't tell people to build every time at each step, but I did leave the non-Windows commands in. What do you think? Do we need to tell people to build after every step, or would a C++ programmer know that?

Comment (tronical) [reply docs/tutorial/cpp/src/memory_tile.md:32]: Hm, the instructions here aren't about building, they're part of the initial setup phase for fetching the icons package. Those differ quite a bit between Windows and macOS/Linux.

For building the instructions are always the same IIRC, just the way of launching is different - and you kept that difference.

So it's just these setup instructions where I'm curious why you remove the Windows specific bits (powershell usage).

File: docs/tutorial/cpp/src/getting_started.md (hunk 4 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/getting_started.md]: @tronical Ahhh, I see. Hmm, I did have to delete build artefacts every time, but I wasn't sure if that was a normal thing to have to do with C++ anyway 😅 . I don't mind, really. If you think it's likely less reliable not to rename it, then I can leave it. I'd rather people had a smooth first Slint experience than get bogged down in C++ issues.

File: docs/tutorial/cpp/src/memory_tile.md (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/tutorial/cpp/src/memory_tile.md:32]: @tronical Purely and oversight, brought them back

File: docs/tutorial/cpp/src/getting_started.md (hunk 4 above)
Comment (tronical) [reply docs/tutorial/cpp/src/getting_started.md]: Yep, that's exactly my worry. I'm leaning towards taking the safe route for now (and not renaming) - the lower the risk the higher the chance of a smooth first experience.

File: docs/tutorial/cpp/src/memory_tile.md (hunk 2 above)
Comment (tronical) [reply docs/tutorial/cpp/src/memory_tile.md:32]: Thanks :)

**Comment by ChrisChinchilla** [comment]
@tronical I think that undid the rename…

**Review by tronical — APPROVED** [review]
